src/opening-period/page/CreateNewOpeningPeriodPage.tsx

```diff
@@ -0,0 +1,228 @@
+  const [periodEndDate, setPeriodEndDate] = useState<Date | null>(null);
+  const [submitStatus, setSubmitStatus] = useState<SubmitStatus>('init');
+
+  const onSubmit = async (data: Inputs): Promise<void> => {
+    try {
+      if (!resource) {
+        throw new Error('This should never happen, but typescript');
+      }
+      await api.postDatePeriod({
+        resource: resource.id,
+        name: {
+          fi: data.openingPeriodTitle,
+          sv: null,
+          en: null,
+        },
+        description: {
+          fi: data.openingPeriodOptionalDescription,
+          sv: null,
+          en: null,
+        },
+        start_date: formatDate(
+          parse(data.openingPeriodBeginDate, 'dd.MM.yyyy', new Date()),
+          'yyyy-MM-dd'
+        ),
+        end_date: formatDate(
+          parse(data.openingPeriodEndDate, 'dd.MM.yyyy', new Date()),
+          'yyyy-MM-dd'
+        ),
+        resource_state: ResourceState.OPEN,
+        override: false,
+        time_span_groups: [],
+      });
+      setSubmitStatus('succeeded');
+    } catch (err) {
+      setSubmitStatus('error');
+      throw err;
+    }
+  };
+
+  useEffect((): void => {
+    // UseEffect's callbacks are synchronous to prevent a race condition.
+    // We can not use an async function as an useEffect's callback because it would return Promise<void>
+    api
+      .getResource(resourceId)
+      .then((r: Resource) => {
+        setResource(r);
+        setLoading(false);
+      })
+      .catch((e: Error) => {
+        setLoadingResourceError(e);
+        setLoading(false);
+      });
+  }, [resourceId]);
+
+  if (isLoading) {
+    return (
+      <>
+        <p>Sivua ladataan...</p>
+      </>
+    );
+  }
+
+  if (hasLoadingResourceError) {
+    return (
+      <>
+        <h1 className="resource-info-title">Virhe</h1>
+        <Notification
+          label="Toimipisteen tietoja ei saatu ladattua."
+          type="error">
+          Tarkista toimipiste-id.
+        </Notification>
+      </>
+    );
+  }
+
+  return (
+    <>
+      {submitStatus === 'succeeded' && (
+        <Notification
+          dataTestId="opening-period-added-successfully-notification"
+          position="top-right"
+          autoClose
+          size="small"
+          label="Aukiolojakso tallennettu onnistuneesti"
+          onClose={(): void => setSubmitStatus('init')}
+          type="success">
+          Aukiolojakso tallennettu onnistuneesti
+        </Notification>
+      )}
+      {submitStatus === 'error' && (
+        <Notification
+          position="top-right"
+          autoClose
+          size="small"
+          label="Aukiolojakson lisääminen epäonnistui"
+          onClose={(): void => setSubmitStatus('init')}
+          type="error">
+          Aukiolojakson lisääminen epäonnistui. Yritä myöhemmin uudestaan.
+        </Notification>
+      )}
+      <div className="add-new-opening-period-page-header-row">
```

Comment:
Naming suggestion: "add-new" -> "new"?
Or would just "opening-period-page-..." be enough?

Reply:
Good suggestion. I think in this case "opening-period-page-header-row" could be sufficient. Since these same styles will also probably be applied in the 'modify opening period page', it fits better for both purposes